src/components/Swap/Input.tsx

```diff
@@ -80,26 +83,33 @@ export default function Input({ disabled, focused }: InputProps) {
   const isDependentField = !useIsSwapFieldIndependent(Field.INPUT)
   const isLoading = isRouteLoading && isDependentField
 
+  const amount = useFormattedFieldAmount({
+    currencyAmount: tradeCurrencyAmount,
+    fieldAmount: inputAmount,
+  })
+
   //TODO(ianlapham): mimic logic from app swap page
   const mockApproved = true
 
-  // account for gas needed if using max on native token
-  const max = useMemo(() => {
-    const maxAmount = maxAmountSpend(balance)
-    return maxAmount?.greaterThan(0) ? maxAmount.toExact() : undefined
-  }, [balance])
-
   const insufficientBalance = useMemo(
     () =>
       balance &&
       (inputCurrencyAmount ? inputCurrencyAmount.greaterThan(balance) : tradeCurrencyAmount?.greaterThan(balance)),
     [balance, inputCurrencyAmount, tradeCurrencyAmount]
   )
 
-  const amount = useFormattedFieldAmount({
-    currencyAmount: tradeCurrencyAmount,
-    fieldAmount: inputAmount,
-  })
+  const max = useMemo(() => {
+    // account for gas needed if using max on native token
+    const max = maxAmountSpend(balance)
+    if (!max || !balance) return
+    if (max.equalTo(0) || balance.lessThan(max)) return
+    if (inputCurrencyAmount && max.equalTo(inputCurrencyAmount)) return
+    return max.toExact()
+  }, [balance, inputCurrencyAmount])
+  const onClickMax = useCallback(() => {
+    invariant(max)
```

Comment:
idk if we use this elsewhere in the codebase but i had to look up what this does because it wasn't immediately obvious to me this was just a check for a truthy value